src/api/authorization-api.ts

```diff
@@ -0,0 +1,40 @@
+import { BASE_API_URL_PATH } from 'core/constants';
+import { type IRequestOptionsData } from 'core/models';
+import HTTPTransport from 'core/services/http-transport';
+
+const authorizationApiInstance = new HTTPTransport();
+const authorizationApiUrlPath = BASE_API_URL_PATH + '/auth';
+
+export interface ISignUp extends IRequestOptionsData {
+    first_name: string;
+    second_name: string;
+    login: string;
+    email: string;
+    password: string;
+    phone: string;
+}
+
+export interface ISignIn extends IRequestOptionsData {
+    login: string;
+    password: string;
+}
+
+class AuthorizationAPI {
+    async signUp(data: ISignUp): Promise<XMLHttpRequest> {
+        return await authorizationApiInstance.post(`${authorizationApiUrlPath}/signup`, { data });
+    }
+
+    async signIn(data: ISignIn): Promise<XMLHttpRequest> {
+        return await authorizationApiInstance.post(`${authorizationApiUrlPath}/signin`, { data });
+    }
+
+    async getUserInfo(): Promise<XMLHttpRequest> {
+        return await authorizationApiInstance.get(`${authorizationApiUrlPath}/user`, {});
```

Comment:
Можно не указывать async await, если функция только возвращает асинхронную функцию, и так вернется промис
```suggestion
    getUserInfo(): Promise<XMLHttpRequest> {
        return authorizationApiInstance.get(`${authorizationApiUrlPath}/user`, {});
```

Reply:
eslint ругается без async await eslint[@typescript-eslint/promise-function-async](https://typescript-eslint.io/rules/promise-function-async).
Что лучше сделать, прописать в линтере более лояльные настройки?

Reply:
Тут на твое усмотрение, как удобнее